aptos-move/block-executor/src/executor.rs

```diff
@@ -655,32 +712,36 @@ where
         txn_idx: TxnIndex,
         versioned_cache: &MVHashMap<T::Key, T::Tag, T::Value, DelayedFieldID>,
         last_input_output: &TxnLastInputOutput<T, E::Output, E::Error>,
+        is_v2: bool,
     ) -> Result<bool, PanicError> {
         let read_set = last_input_output
             .read_set(txn_idx)
             .expect("Read set must be recorded");
 
-        let mut execution_still_valid =
-            read_set.validate_delayed_field_reads(versioned_cache.delayed_fields(), txn_idx)?;
+        if !read_set.validate_delayed_field_reads(versioned_cache.delayed_fields(), txn_idx)?
+            || (is_v2
+                && !read_set.validate_aggregator_v1_reads(
+                    versioned_cache.data(),
+                    last_input_output.modified_aggregator_v1_keys(txn_idx),
```

Comment:
If this is None, it means txn aborted or there is speculative failure, is it even possible? We should return an error on `None`? I see that `check_execution_status_during_commit` does check for this but after validation at commit.